bld/cb.cs

```diff
@@ -355,6 +355,57 @@ IList<string> libs
         }
     }
 
+    static void write_wasm(
+        string libname,
+        IList<string> cfiles,
+        Dictionary<string, string> defines,
+        IList<string> includes,
+        IList<string> libs)
+    {
+        var dest_filelist = string.Format("wasm_{0}.libtoolfiles", libname);
+        using (TextWriter tw = new StreamWriter(dest_filelist))
+        {
+            var subdir = string.Format("{0}/wasm", libname);
+            foreach (var s in cfiles)
+            {
+                var b = Path.GetFileNameWithoutExtension(s);
+                var o = string.Format("./obj/{0}/{1}.o", subdir, b);
+                tw.Write("{0}\n", o);
+            }
+        }
+        using (TextWriter tw = new StreamWriter(string.Format("wasm_{0}.sh", libname)))
+        {
+            tw.Write("#!/bin/sh\n");
+            tw.Write("set -e\n");
+            tw.Write("set -x\n");
+            var subdir = string.Format("{0}/wasm", libname);
+            tw.Write("mkdir -p \"./obj/{0}\"\n", subdir);
+            foreach (var s in cfiles)
+            {
+                tw.Write("emcc");
+                tw.Write(" -Oz");
```

Comment:
NB: Using `-Oz` instead of `-O` here since that's what the mono-wasm SDK does:

https://github.com/dotnet/runtime/blob/50e6f078bcf4aa5bca1e7e110b9a6cb846390b82/src/mono/wasm/build/WasmApp.Native.targets#L172-L174